src/rpcLoggingProvider.ts

```diff
@@ -1,17 +1,52 @@
 import winston from "winston";
 import { Provider as ZkSyncProvider } from "zksync-ethers";
+import { IBridgehub__factory } from "zksync-ethers/build/typechain";
+
+import type { Provider as EthersProvider } from "ethers";
+import type { Fee, TransactionRequest } from "zksync-ethers/build/types";
 
 /**
  * Custom Provider wrapper that logs all JSON-RPC calls
  */
 export class LoggingZkSyncProvider extends ZkSyncProvider {
   private requestId: number = 1;
+  private l1Provider: EthersProvider | null = null;
+  private isZKsyncOS = false;
 
   constructor(url: string) {
     // Pass the URL to the parent class constructor
     super(url);
   }
 
+  setIsZKsyncOS(isZKsyncOS: boolean) {
+    this.isZKsyncOS = isZKsyncOS;
+  }
+
+  setL1Provider(l1Provider: EthersProvider) {
+    this.l1Provider = l1Provider;
+  }
+
+  override async getBaseTokenContractAddress(): Promise<string> {
```

Comment:
comment please
```suggestion
  /// method overriden to use L1 calls instead of zks_ method for compatbility with ZK OS
  override async getBaseTokenContractAddress(): Promise<string> {
```